### ISSUE TITLE
[HOLD for payment 2024-02-15] [LOW] [$500] Mweb/Chrome - Status - In status, emoji selected is not shown in preview as selected

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#33679]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.18-2
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause - Internal Team
**Slack conversation:**

## Action Performed:
1. Go to https://staging.new.expensify.com/
2. Tap profile icon --- Profile
3. Tap status
4. Tap emoji picker
5. Select an emoji

## Expected Result:
In status, emoji selected must be shown in preview as selected

## Actual Result:
In status, emoji selected is not shown in preview as selected

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [x] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/Expensify/App/assets/78819774/3e7cc161-c073-40a2-b4f2-07e037c81c33

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~018c9ea15b7ecaf817</li>
        <li>Upwork Job ID: 1740170921969528832</li>
        <li>Last Price Increase: 2024-01-18</li>
<li>Automatic offers: </li>
<ul>
<li>getusha | Reviewer | 28121816</li>
<li>wildan-m | Contributor | 28121817</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @strepanier03</details>

**Comment by shubham1206agra** [comment]
@lanitochka17 Unable to repro this. Is there any special steps which I am missing?

**Comment by bernhardoj** [comment]
I think @lanitochka17 expects the status emoji is highlighted/focused on the emoji picker which sounds like a new feature

**Comment by strepanier03** [comment]
> I think @lanitochka17 expects the status emoji is highlighted/focused on the emoji picker which sounds like a new feature

I agree with you @bernhardoj, that does appear to be the expectation. I'll confirm internally if this is a good fit for one of our waves and update this once I have more info. 

Note: We have a lot of folks OoO for the holidays so I don't expect an update until next week.

**Comment by strepanier03** [comment]
I bumped internally because I think we can either include this in a VIP as a new feature or we can close this entirely. 

I tested in Slack and the behavior in Slack is the same as the behavior here, where the emoji isn't selected, so at this time I'm leaning towards closing but will update when I know for sure.

**Comment by strepanier03** [comment]
This is a LOW priority issue for the VIP-VSB.

**Comment by strepanier03** [comment]
Low priority for now unless someone posts a proposal for review.

**Comment by wildan-m** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

Status - In status, emoji selected is not shown in preview as selected

### What is the root cause of that problem?

No such feature yet

### What changes do you think we should make in order to solve the problem?
We can pass `activeEmoji` to EmojiPicker through `showEmojiPicker`and compare with each emoji.

If required, we can ignore skin tone by comparing its base emoji using this function
```
const getRemovedSkinToneEmoji = (emoji: string) => {
    return emoji.replace(CONST.REGEX.EMOJI_SKIN_TONES, '');
};
```

Where `EMOJI_SKIN_TONES` is `/[\u{1f3fb}-\u{1f3ff}]/gu`

To highlight the emoji we can modify this logic

https://github.com/Expensify/App/blob/176fb0011905c477c4e8ef3bb9ef472df4ae46d9/src/components/EmojiPicker/EmojiPickerMenu/index.js#L347

to

```
            const shouldEmojiBeHighlighted =
                (index === highlightedIndex && highlightFirstEmoji) || (activeEmoji && EmojiUtils.getRemovedSkinToneEmoji(emojiCode) === EmojiUtils.getRemovedSkinToneEmoji(activeEmoji));

```

If we want to consider skin tone (exact match) we can use `emojiCode === activeEmoji` instead


[Branch to test](https://github.com/Expensify/App/compare/main...wildan-m:App:wildan/fix/33679-highlight-active-emoji?expand=1).

### What alternative solutions did you explore? (Optional)

**Comment by getusha** [comment]
Will review proposals.

**Comment by strepanier03** [comment]
Thanks @getusha!

**Comment by strepanier03** [comment]
@getusha - Any update on the proposals you reviewed?

**Comment by strepanier03** [comment]
Not sure what's up with the Overdue label today 🤷‍♀️

**Comment by strepanier03** [comment]
Starred this on myself hoping that works.

**Comment by getusha** [comment]
Testing @wildan-m's [proposal](https://github.com/Expensify/App/issues/33679#issuecomment-1889553403) will try to provide an update sooner.

**Comment by wildan-m** [comment]
Thanks @getusha

**Comment by getusha** [comment]
@wildan-m's proposal appears to work to highlight the emoji in the Emojipicker, i am not sure if we want to do that. i'll open a discussion in `expensify-open-souce`

**Comment by getusha** [comment]
Started a discussion
https://expensify.slack.com/archives/C01GTK53T8Q/p1706117484845689

**Comment by getusha** [comment]
Let's proceed with @wildan-m's [proposal](https://github.com/Expensify/App/issues/33679#issuecomment-1889553403), and we can address any other cases as they arise during the PR review.
🎀 👀 🎀  C+ reviewed!

**Comment by wildan-m** [comment]
@getusha, @hayata-suenaga in case we want to proceed, I have prepared [a PR ready for review](https://github.com/Expensify/App/pull/35140).

**Comment by getusha** [comment]
@wildan-m you should not raise a PR before assignment.

**Comment by wildan-m** [comment]
@getusha no worries we can just close it if it's not required, I have limited availability in the next hours, so it's just optimistic scenario

**Comment by wildan-m** [comment]
@hayata-suenaga, @getusha awesome! thanks for assigning me.

**Comment by jjcoffee** [comment]
Ideally I think we should hold this on https://github.com/Expensify/App/pull/34581 (will hopefully be merged soon).

**Comment by hayata-suenaga** [comment]
@jjcoffee let me know when the PR you linked is merged so that I can remove the hold

**Comment by wildan-m** [comment]
@jjcoffee thanks for letting us know.


@getusha The PR conflict has been resolved. [PR Link](https://github.com/Expensify/App/pull/35140)

**Comment by hayata-suenaga** [comment]
yaaay! @getusha could you review the PR when you have time?

**Comment by getusha** [comment]
@hayata-suenag will review it.

**Comment by wildan-m** [comment]
@strepanier03 @getusha @hayata-suenaga seems no regression here(?)

**Comment by hayata-suenaga** [comment]
Yes I think the regression wait period is over
@wildan-m please wait for others to complete the checklist 🙇

**Comment by getusha** [comment]
BugZero Checklist: The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

 [@getusha] The PR that introduced the bug has been identified. Link to the PR: **new feature**
 [@getusha] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: **new feature**
 [@getusha] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion: **N/a it is a new feature**
 [@getusha] Determine if we should create a regression test for this bug. **Yes**
 [@getusha] If we decide to create a [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) for the bug, please propose the regression test steps to ensure the same bug will not reach production again.

**Regression Test Proposal**

1. Go to account settings > Profile > Status
2. Pick a new emoji
3. Open the emoji picker
4. Verify that the picked emoji is highlighted inside the Emojipicker.

Do we agree 👍 or 👎

**Comment by hayata-suenaga** [comment]
@strepanier03 please add the above new test when you have time 🙇
